hashes/src/lib.rs

```diff
@@ -219,6 +217,24 @@ pub trait Hash:
     fn all_zeros() -> Self;
 }
 
+/// Attempted to create a hash from an invalid length slice.
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub struct FromSliceError {
+    expected: usize,
+    got: usize,
```

Comment:
Might be useful to make these `pub` and make the struct `#[non_exhaustive]`.

Reply:
Made mental note to play with this idea and check usability of such an error type. Thanks